src/components/atoms/BaseCard/BaseCard.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, KeyboardEvent } from 'react';
+import { createUseStyles } from 'react-jss';
+
+/**
+ * Properties for BaseCard component
+ */
+export interface BaseCardProps {
+  /** Element to show at card front face */
+  frontFace?: JSX.Element;
+  /** Element to show at card back face */
+  backFace?: JSX.Element;
+  /** Defines the card height */
+  height?: number;
+  /** Defines the card width */
+  width?: number;
+}
+
+const useStyles = createUseStyles({
+  // Generic classes
+  cardFace: {
+    backfaceVisibility: 'hidden',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    color: 'cornSilk',
+    textAlign: 'center',
+    font: `3em/240px 'Helvetica Neue', Helvetica, sans-serif`,
+    boxShadow: '-5px 5px 5px #aaa',
+    transition: '0.6s',
+  },
+  // Modal classes
+  backVisible: {},
+  frontVisible: {},
+  // Style classes
+  card: ({ width, height }) => ({
+    width: width || 600,
+    height: height || 400,
+    perspective: 1000,
+    cursor: 'pointer',
+    '&$backVisible $cardFlipper': { transform: 'rotateY(180deg)' },
+    '&$backVisible $cardFront': { opacity: 0 },
+    '&$backVisible $cardBack': { opacity: 1 },
+    '&$frontVisible $cardFront': { opacity: 1 },
+    '&$frontVisible $cardBack': { opacity: 0 },
+    outline: 'none',
+    overflow: 'hidden',
+  }),
+  cardFlipper: {
+    transition: '0.6s',
+    transformStyle: 'preserve-3d',
+    position: 'relative',
+  },
+  cardFront: {
+    extend: 'cardFace',
+    zIndex: 2,
+    transform: 'rotateY(0deg)',
+  },
+  cardBack: {
+    extend: 'cardFace',
+    transform: 'rotateY(180deg)',
+  },
+});
+
+/**
+ * Base card molecule
+ * @param props Card props
+ */
+export const BaseCard: React.FC<BaseCardProps> = ({
+  frontFace,
+  backFace,
+  width = undefined,
+  height = undefined,
+} = {}) => {
+  const { backVisible, frontVisible, card, cardFlipper, cardFront, cardBack } = useStyles({ width, height });
+  const [visibleFace, setVisibleFace] = useState<'front' | 'back'>('front');
```

Comment:
so regarding my comment from before it's ok to keep it as string I guess but the problem to me is that you hardcode them. It would be better if you'd use a const for that like 
``` 
const frontFace = 'front';
const backFace = 'back';
type VisibleFace = frontFace | backFace;
```
and then later on

```
const [visibleFace, setVisibleFace] = useState<VisibleFace>(frontFace);
```

and so on, problem here is that if at some point we decide to rename the 'front' string into 'frontFace' for instance, you'd have to change it everywhere in the code instead of changing it in one place only.

Reply:
So, by your suggestion, it would become:

```typescript
const frontLabel = 'front';
const backLabel = 'back';
type VisibleFace = typeof frontLabel | typeof backLabel;

export const BaseCard: React.FC<BaseCardProps> = ({
  frontFace,
  backFace,
  width = undefined,
  height = undefined,
} = {}) => {
       const [visibleFace, setVisibleFace] = useState<VisibleFace>(frontFace);
       const turnCard = (): void => setVisibleFace(visibleFace === frontLabel ? backLabel : frontLabel);

...
```
Don't you feel it a bit unnatural?
I mean, in general I am totally inline for not hardcoding because of two things:

1 - If you need to rewrite on several places
2 - if you forget to replace somewhere, you may break the app without knowing (like in JS without type checking)

In here we are talking about:
1 - a single file with less then 100 lines
2 - if you forget to replace somewhere, your build will fail and will tell you exact the lines where you should be updating.

I don't know, never seen this kind of approach in typescript
We can discuss over our talk later